### ISSUE TITLE
What is the correct method of using it in Flutter 3.10?

### ISSUE THREAD

**Issue by fzyzcjy** [getsentry/sentry-dart#1482]
https://pub.dev/packages/sentry says:

![image](https://github.com/getsentry/sentry-dart/assets/5236035/0f0fdf2f-7a30-44a8-aad7-219ab28c4706)

However, https://docs.flutter.dev/release/breaking-changes i.e. https://docs.flutter.dev/release/breaking-changes/zone-errors says:

> Starting with Flutter 3.10, the framework detects mismatches when using Zones and reports them to the console in debug builds.
> ... calls to WidgetsFlutterBinding.ensureInitialized() should be run in the same zone as calls to runApp().

EDIT: I see, the appRunner is no longer executed within a zone (unlike the old days), so the doc is already correct.

**Comment by marandaneto** [comment]
You don't need to set up your own Zone (runZonedGuard) or the new error callback, you also don't need to call `ensureInitialized`.
You can though, but then you are responsible to call `captureException` yourself, the docs are up to date, yes.

**Comment by Ciock** [comment]
I'm sorry to comment on a closed issue, but I got the `Zone mismatch` error with `sentry_flutter: ^7.6.2`.


``` dart
Future<void> initAppAndRun() async {
  await SentryFlutter.init(
    (options) {
      options.enableTracing = true;
      options.dsn = 'https://xxxx';
      options.environment = AppService.to.environment.name;
      options.attachScreenshot = true;
      options.screenshotQuality = SentryScreenshotQuality.low;
      options.tracesSampleRate = 0.1;
    },
    appRunner: () async {
      WidgetsFlutterBinding.ensureInitialized();

      // Removes '#' from url
      setPathUrlStrategy();

      SystemChrome.setPreferredOrientations(
        [DeviceOrientation.portraitUp, DeviceOrientation.portraitDown],
      );

      await Firebase.initializeApp(˙
        options: DefaultFirebaseOptions.currentPlatform,
      );
      FirebaseMessagingService.initListeners();

      DioService.to.dio.addSentry();

      runApp(const MyApp());
    },
  );
}

```

I got no error if I remove the Sentry code:



``` dart
Future<void> initAppAndRun() async {
  WidgetsFlutterBinding.ensureInitialized();

  // Removes '#' from url
  setPathUrlStrategy();

  SystemChrome.setPreferredOrientations(
    [DeviceOrientation.portraitUp, DeviceOrientation.portraitDown],
  );

  await Firebase.initializeApp(˙
    options: DefaultFirebaseOptions.currentPlatform,
  );
  FirebaseMessagingService.initListeners();

  runApp(const MyApp());
}
```

P.S. I use different main files to set many environments, this is why the method is not called `main`.

**Comment by memishood** [comment]
I m facing with same issue on flutter web

**Comment by marandaneto** [comment]
Raise a new issue if you are running into issues.

TLDR: You don't need to call `WidgetsFlutterBinding.ensureInitialized();` at all, Sentry already does that internally.
